crates/solidity/inputs/language/src/definition.rs

```diff
@@ -2865,14 +2858,17 @@ codegen_language_macros::compile!(Language(
                         Enum(
                             name = Statement,
                             variants = [
-                                EnumVariant(
-                                    name = TupleDeconstruction,
-                                    reference = TupleDeconstructionStatement
-                                ),
+                                // Simple statements
+                                EnumVariant(name = Expression, reference = ExpressionStatement),
```

Comment:
I see `ExpressionStatement` was moved to the top. Is it just for organization, or does it impact parsing?

I'm asking because almost all other variants start with a unique token, so it should be faster to parse/validate them, instead of trying to parse a full expression, then backtracking when it fails.

Reply:
Same question for `YulStatement`

Reply:
It's in the OP:

> - Syncs ordered choice for `(Yul)Statement` across DSLs (helps minimize diff for #650)

The error recovery attempts to pick a best parse, so changing the ordered choice will change the resulting recovered parse, given that the alternatives cover the same amount of valid bytes. Using the new definition changes the resulting CST in one of our tests it seems and I wanted to bring it back for now.

In general, I understand and see value in trying to speed up the parse, but I think we shouldn't introduce accidental differences like that as long as we have 3 different language definitions. Once we move to a single definition, I'm happy to come back and work on measuring and improving the performance, especially since we didn't even start picking low-hanging fruits such as limiting the backtracking in general.

Reply:
Makes sense.